### PR TITLE
Fix #3674: Added new test cases in PlatformParameterIntegrationTest

### LINKED ISSUE #3674: Include tests for SyncUp mechanism in the Integration Test for Platform Parameters
Update PlatformParameterIntegrationTest to include test based on refreshing with SyncUpWorker.

Target PR date: 11 August 2021
Target completion date: 16 August 2021

### PR DESCRIPTION
**Pull request by Arjupta**
## Explanation

Fixed #3674 
Updated PlatformParameterIntegrationTest to include test based on refreshing with SyncUpWorker.

Target PR date: 11 August 2021
Target completion date: 16 August 2021
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by anandwana001** [comment]
Left some thoughts over chat, let's see if it works or not. 

>in my understanding, the UI test which requires this much amount for pre setting of data, then it should be shifted to unit test only. 
And for UI we can test by updating the data thing over testActivity files.

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (jcqli) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Why do we use this instead of @Inject for the context?

File: testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt
Comment (jcqli) [inline testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt]: Recommend removing the comment and changing the `TEST_OBJECT_AS_PARAM_VALUE` => `TEST_UNSUPPORTED_OBJECT_AS_PARAM_VALUE`

File: testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt
Comment (jcqli) [inline testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt]: Can probably remove this comment.

**Review by jcqli — CHANGES_REQUESTED** [review]
Overall looks pretty good, although definitely want some more feedback from others about the test.

File: testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt
Comment (Arjupta) [reply testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt]: Done

File: testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt
Comment (Arjupta) [reply testing/src/main/java/org/oppia/android/testing/network/MockPlatformParameterService.kt]: Yeah, Done.

File: app/src/main/java/org/oppia/android/app/testing/SplashTestActivityPresenter.kt
Comment (vinitamurthi) [inline app/src/main/java/org/oppia/android/app/testing/SplashTestActivityPresenter.kt:34]: Add a KDoc for this function now that you have opened it up!

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (vinitamurthi) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Maybe I didnt completely understand -- why are we updating this here with an empty list? Dont we want the DB to have some initial values? Or if you are using the initial value as empty, lets instead have the initial value explicitly as false

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Changed it to @Inject format. I was debugging with InstrumentationRegistry I guess.

File: app/src/main/java/org/oppia/android/app/testing/SplashTestActivityPresenter.kt
Comment (Arjupta) [reply app/src/main/java/org/oppia/android/app/testing/SplashTestActivityPresenter.kt:34]: Done

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Yeah we can have a false value in the database which will be better proof that the values changed.

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (jcqli) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt:134]: nit: Can this fit on a a single line?
```suggestion
    listOf<PlatformParameter>(mockSplashScreenWelcomeMsgParam)
```

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (jcqli) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Can this fit on a a single line?
```suggestion
    listOf<PlatformParameter>(mockSplashScreenWelcomeMsgParam)
```

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (jcqli) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: For clarity, probably can do:
```suggestion
    launch(SplashTestActivity::class.java).use { scenario ->
```

and also use `activity` below, ditto elsewhere.

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (jcqli) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]:
```suggestion
      scenario.onActivity { activity.splashTestActivityPresenter.loadPlatformParameters() }
```

**Review by jcqli — APPROVED** [review]
A few more comments for clarity but otherwise LGTM!

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Done

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Done

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Done

**Review by vinitamurthi — APPROVED** [review]

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: shouldn't be the name says `setUpApplicationForVersionName` ?

**Review by anandwana001 — APPROVED** [review]
Nit change, approving the PR, test LGTM

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Yeah. Done

File: utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt]: Please add KDocs for these values.

File: app/BUILD.bazel
Comment (BenHenning) [inline app/BUILD.bazel]:
```suggestion
    "//testing/src/main/java/org/oppia/android/testing/network",
```

Simplification.

**Review by BenHenning — COMMENTED** [review]
Thanks @ARJUPTA. Just had a couple of comments per my codeowner files.

File: utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt
Comment (Arjupta) [reply utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt]: Done

File: app/BUILD.bazel
Comment (Arjupta) [reply app/BUILD.bazel]: Done

**Comment by BenHenning** [comment]
@ARJUPTA I unfortunately didn't review this before merging the base PR, and trying to review the deltas isn't working very well. Can you please bring this up-to-date with develop & properly incorporate the base deltas so that the PR changes are correct?

**Comment by Arjupta** [comment]
> @ARJUPTA I unfortunately didn't review this before merging the base PR, and trying to review the deltas isn't working very well. Can you please bring this up-to-date with develop & properly incorporate the base deltas so that the PR changes are correct?

@BenHenning I have updated the PR with develop. You can see your requested changes now

File: utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt]: welcome message

Ditto in other KDocs.

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt:222]: This flow seems a bit odd; shouldn't the app be set up before launching the test activity?

**Review by BenHenning — COMMENTED** [review]
Thanks @ARJUPTA. Just 2 nits otherwise the PR LGTM.

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt:222]: We are doing so because we needed to change the app version for testing the work request for different network response. But if we were doing so before the activity is launched then the SplashTestActivity was not able to be launched by tests. @anandwana001, @vinitamurthi  and @jcqli are known of this issue but thre is no such fix for this as based on our research. Also even if we are doing it this way, we actually fetch the values after the work request is completed. Therefore it still checks the integration correctly

File: utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt
Comment (Arjupta) [reply utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt]: Done

File: app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (BenHenning) [reply app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt:222]: I'd need more info on what specifically is preventing us from doing the other approach, but given this is a test arrangement I'm not too bothered by this.

**Review by BenHenning — APPROVED** [review]
Thanks @ARJUPTA. LGTM barring CI checks.

Edit: also, given that the previous run passed on Bazel checks I'm going to enable auto-merge since the deltas are just comment changes.

**Pull request merged**
